Update Travis CI config

gavanderhoorn opened the pull request:
Changes:

 - don't allow Melodic jobs to fail
 - switch to `master` branch

Enabling `ccache` won't really help that much, as only `ur_kinematics` now still contains C++, but it won't hurt.

gavanderhoorn committed:
travis: switch to master.

gavanderhoorn committed:
travis: Melodic jobs should pass.

We no longer allow those to fail.

gavanderhoorn commented:
@ipa-mdl: if you could take a look and confirm I'm not doing anything stupid here, that would be appreciated.

gavanderhoorn committed:
travis: shadow-fixed is now testing.

mathias-luedtke commented:
> switch to master branch

Not necessary anymore :sunglasses:

gavanderhoorn commented:
> > switch to master branch
> 
> Not necessary anymore sunglasses

Still like to be explicit about this. It's very clear that this config is using `master`.

And we don't run the risk of you changing default branches on us ;)

mathias-luedtke commented on .travis.yml:
```diff
@@ -2,24 +2,27 @@ language: generic
-    - env: ROS_DISTRO="melodic"   ROS_REPO=ros-shadow-fixed
-    - env: ROS_DISTRO="melodic"   PRERELEASE=true
+    - ROS_DISTRO="kinetic"
+    - ROS_DISTRO="kinetic" ROS_REPO=ros-testing
```
`ROS_REPO=[ros-]testing` is the default, you would run the same job twice.
To use the stable release, you have to set `ROS_REPO=ros` (even better: `ROS_REPO=main`) explicitly.
If you want to, you can mark the others with `ROS_REPO=testing` for readability.

gavanderhoorn replied:
Isn' that done by the `ROS_REPO=ros` in the `global` section?

mathias-luedtke replied:
Ah yes, you are right. I did not see that line.
I would remove it as a global. You are adding both variants anyway.

mathias-luedtke commented on .travis.yml:
```diff
@@ -2,24 +2,27 @@ language: generic
+
 install:
-  - git clone --quiet --depth=1 https://github.com/ros-industrial/industrial_ci.git .ci_config
+  - git clone --quiet --depth=1 -b master https://github.com/ros-industrial/industrial_ci.git .ci_config
```
Nitpick: I never understood why it was named `.ci_config`, I have updated all documentation to `.industrial_ci`.

gavanderhoorn committed:
travis: it's actually the 'main' repository.

gavanderhoorn committed:
travis: update local name for cloned industrial_ci repo.

Old name probably cargo-cult from way-back-when.

gavanderhoorn committed:
travis: set repo on matrix row.

Instead of as a global.

Having it as part of the row increases visibility and avoids confusion.

gavanderhoorn commented:
Just noticed: this is the 500th ~~PR~~ PR or issue on this repository :clinking_glasses:

mathias-luedtke commented on .travis.yml:
```diff
@@ -2,24 +2,26 @@ language: generic
-    - env: ROS_DISTRO="melodic"   ROS_REPO=ros-shadow-fixed
-    - env: ROS_DISTRO="melodic"   PRERELEASE=true
+    - ROS_DISTRO="kinetic" ROS_REPO=main
+    - ROS_DISTRO="kinetic" ROS_REPO=ros-testing
```
`ROS_REPO=testing` in the new naming convention..

`ros-testing` is the actually name of the ROS1 repository on the server..
For now it is treated like `testing`, but this might change in the future.

gavanderhoorn replied:
I changed that based on a notice from `industrial_ci` in the first job that was run I believe.

Anyway, I'll change it.

mathias-luedtke replied:
https://github.com/ros-industrial/industrial_ci/blame/master/industrial_ci/src/env.sh#L91 ?

gavanderhoorn replied:
Probably, I cannot find it in the build log any more.

I must have misremembered.

gavanderhoorn committed:
travis: use correct name for 'testing' repo.

gavanderhoorn commented:
Thanks for the review @ipa-mdl :+1:
